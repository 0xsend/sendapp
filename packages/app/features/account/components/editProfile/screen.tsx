import {
  Button,
  Checkbox,
  FadeCard,
  Paragraph,
  ProfileAvatar,
  Separator,
  Spinner,
  SubmitButton,
  XStack,
  YStack,
} from '@my/ui'
import { SchemaForm } from 'app/utils/SchemaForm'
import { ProfileSchema, useProfileMutation } from 'app/utils/useProfileMutation'
import { useUser } from 'app/utils/useUser'
import { UploadAvatar, type UploadAvatarRefObject } from '../uploadProfileImage/screen'
import { useRef, useState } from 'react'
import type { Tables } from '@my/supabase/database.types'
import { Check } from '@tamagui/lucide-icons'
import { SettingsHeader } from 'app/features/account/settings/components/SettingsHeader'
import { ReadOnlyFieldWithLabel } from 'app/features/account/settings/components/ReadOnlyFieldWithLabel'
import { FieldWithLabel } from 'app/features/account/settings/components/FieldWithLabel'
import { useForm } from 'react-hook-form'
import type { z } from 'zod'

enum FormState {
  Overview = 'Overview',
  ProfileForm = 'ProfileForm',
}

export const EditProfile = () => {
  const [formState, setFormState] = useState<FormState>(FormState.Overview)
  const { profile } = useUser()

  return (
    <YStack w={'100%'}>
      <YStack gap={'$3.5'}>
        <SettingsHeader>Profile</SettingsHeader>
        {(() => {
          switch (true) {
            case !profile:
              return <Spinner size="large" />
            case formState === FormState.Overview:
              return (
                <Overview profile={profile} onPress={() => setFormState(FormState.ProfileForm)} />
              )
            case formState === FormState.ProfileForm:
              return (
                <EditProfileForm
                  profile={profile}
                  onSave={() => setFormState(FormState.Overview)}
                />
              )
          }
        })()}
      </YStack>
    </YStack>
  )
}

const Overview = ({ profile, onPress }: { profile: Tables<'profiles'>; onPress: () => void }) => {
  const { name, about, is_public, avatar_url } = profile
  const avatarRef = useRef<UploadAvatarRefObject>(null)

  return (
    <YStack gap={'$5'}>
      <FadeCard>
        <XStack gap={'$5'} width={'100%'}>
          <UploadAvatar ref={avatarRef}>
            <ProfileAvatar
              avatarUrl={avatar_url ? avatar_url : undefined}
              $gtMd={{ size: 88 }}
              size={88}
            />
          </UploadAvatar>
          <YStack gap={'$2'} ai={'flex-start'}>
            <YStack>
              <Paragraph
                size={'$5'}
                fontWeight={500}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Profile Picture
              </Paragraph>
            </YStack>
            <Button unstyled onPress={() => avatarRef.current?.pickImage()}>
              <Button.Text
                textDecorationLine="underline"
                color="$primary"
                $theme-light={{ color: '$color12' }}
                size={'$5'}
              >
                Update
              </Button.Text>
            </Button>
          </YStack>
        </XStack>
        <FieldWithLabel label={'Name'}>
          <Paragraph size={'$8'} fontWeight={'500'}>
            {name || '-'}
          </Paragraph>
        </FieldWithLabel>
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <ReadOnlyFieldWithLabel label={'About'} text={about || '-'} />
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <XStack gap={'$3'} ai={'center'}>
          <Checkbox
            disabled={true}
            checked={!!is_public}
            borderWidth={0}
            backgroundColor={is_public ? '$primary' : '$background'}
            circular={true}
          >
            <Checkbox.Indicator>
              <Check color={'$black'} />
            </Checkbox.Indicator>
          </Checkbox>
          <Paragraph size={'$5'}>{`${is_public ? 'Public' : 'Private'}`} Profile</Paragraph>
        </XStack>
      </FadeCard>
      <SubmitButton theme="green" borderRadius={'$4'} p={'$4'} onPress={onPress}>
        <Button.Text ff={'$mono'} fontWeight={'500'} tt="uppercase" size={'$5'} color={'$black'}>
          edit profile
        </Button.Text>
      </SubmitButton>
    </YStack>
  )
}

function EditProfileForm({ profile, onSave }: { profile: Tables<'profiles'>; onSave: () => void }) {
  const { id, name, about, is_public } = profile
  const { mutateAsync, error } = useProfileMutation(id)
  const form = useForm<z.infer<typeof ProfileSchema>>()

  const handleSubmit = async () => {
    const values = form.getValues()
    await mutateAsync(values)
    onSave()
  }

  return (
    <SchemaForm
      form={form}
      schema={ProfileSchema}
      props={{
        name: {
          'aria-label': 'Name',
          bc: '$color0',
        },
        about: {
          'aria-label': 'Bio',
          placeholder: 'Tell us about yourself',
          backgroundColor: '$color0',
          rows: 2,
          focusStyle: {
            fontStyle: 'normal',
          },
        },
        isPublic: {
          defaultChecked: is_public !== null ? is_public : true,
        },
      }}
      defaultValues={{
        name: name ? name : '',
        about: about ? about : '',
        isPublic: is_public !== null ? is_public : true,
      }}
      formProps={{
        $gtSm: {
          maxWidth: '100%',
        },
      }}
      onSubmit={handleSubmit}
      renderAfter={({ submit }) => (
        <YStack>
          <SubmitButton
            theme="green"
            borderRadius={'$4'}
            p={'$4'}
            mt={'$1'}
            onPress={() => submit()}
          >
            <Button.Text
              ff={'$mono'}
              fontWeight={'500'}
              tt="uppercase"
              size={'$5'}
              color={'$black'}
            >
              SAVE CHANGES
            </Button.Text>
          </SubmitButton>
          {error && (
            <Paragraph marginTop={'$3'} theme="red" color="$color9">
              {error.message}
            </Paragraph>
          )}
        </YStack>
      )}
    >
      {({ name, about, isPublic }) => (
        <FadeCard>
          <FieldWithLabel label={'Name'} gap={'$2'}>
            {name}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <FieldWithLabel label={'About'} gap={'$2'}>
            {about}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <XStack gap={'$3'} ai={'center'}>
            {isPublic}
            <Paragraph size={'$5'}>Make Profile Public</Paragraph>
          </XStack>
        </FadeCard>
      )}
    </SchemaForm>
  )
}
