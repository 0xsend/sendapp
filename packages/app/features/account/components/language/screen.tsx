import { Card, Paragraph, Separator, Spinner, XStack, YStack } from '@my/ui'
import { SettingsHeader } from 'app/features/account/components/SettingsHeader'
import { useTranslation } from 'react-i18next'
import { useCallback, useMemo, useState } from 'react'
import { Pressable, Platform } from 'react-native'
import { Check } from '@tamagui/lucide-icons'
import { getSupportedLocaleOptions } from 'app/i18n/locales'
import { useHoverStyles } from 'app/utils/useHoverStyles'

export const LanguagePreferences = () => {
  const { i18n, t } = useTranslation('settings')
  const [pendingLocale, setPendingLocale] = useState<string | null>(null)
  const resolvedLanguage = i18n.resolvedLanguage ?? i18n.language

  const options = useMemo(() => getSupportedLocaleOptions(), [])
  const hoverStyles = useHoverStyles()

  const handleSelect = useCallback(
    async (locale: string) => {
      if (locale === resolvedLanguage || pendingLocale === locale) {
        return
      }

      setPendingLocale(locale)
      try {
        await i18n.changeLanguage(locale)
      } finally {
        setPendingLocale(null)
      }
    },
    [i18n, pendingLocale, resolvedLanguage]
  )

  return (
    <YStack gap={'$3.5'} w={'100%'}>
      {Platform.OS === 'web' && <SettingsHeader>{t('language.title')}</SettingsHeader>}
      <Card gap={'$3'} padded size={'$4'}>
        <Paragraph size={'$4'} color={'$color12'}>
          {t('language.description')}
        </Paragraph>
        {pendingLocale ? (
          <Paragraph size={'$2'} color={'$lightGrayTextField'}>
            {t('language.updating')}
          </Paragraph>
        ) : null}
        <Separator boc={'$darkGrayTextField'} opacity={0.2} />
        <YStack>
          {options.map((option, index) => {
            const isActive = option.value === resolvedLanguage
            const isPending = pendingLocale === option.value

            return (
              <YStack key={option.value}>
                <Pressable
                  accessibilityRole="button"
                  accessibilityState={{ selected: isActive, busy: isPending }}
                  disabled={isPending}
                  onPress={() => handleSelect(option.value)}
                >
                  <XStack
                    ai="center"
                    jc="space-between"
                    px={'$2'}
                    py={'$3'}
                    br={'$4'}
                    hoverStyle={hoverStyles}
                  >
                    <Paragraph size={'$5'}>{option.autonym}</Paragraph>
                    {isPending ? (
                      <Spinner size="small" />
                    ) : isActive ? (
                      <Check size={16} color={'$primary'} />
                    ) : null}
                  </XStack>
                </Pressable>
                {index < options.length - 1 && (
                  <Separator boc={'$darkGrayTextField'} opacity={0.15} />
                )}
              </YStack>
            )
          })}
        </YStack>
      </Card>
    </YStack>
  )
}
