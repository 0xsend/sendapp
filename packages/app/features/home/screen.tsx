import {
  Anchor,
  Avatar,
  Button,
  Card,
  Container,
  ListItem,
  Paragraph,
  ScrollView,
  Theme,
  ThemeName,
  XStack,
  YStack,
} from '@my/ui'
import { useThemeSetting } from '@tamagui/next-theme'
import {
  IconArrowDown,
  IconClose,
  IconDeposit,
  IconEthereum,
  IconReceive,
  IconSend,
  IconSendTile,
  IconUSDC,
} from 'app/components/icons'
import { MainLayout } from 'app/components/layout'
import { CommentsTime } from 'app/utils/dateHelper'
import formatAmount from 'app/utils/formatAmount'
import { useState } from 'react'
import { Square } from 'tamagui'
export function HomeScreen() {
  const { resolvedTheme } = useThemeSetting()
  const separatorColor = resolvedTheme?.startsWith('dark') ? '#343434' : '#E6E6E6'
  const USDollar = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  })
  const [expandBalance, setExpandBalance] = useState(false)
  const actionButtons = [
    { label: 'Deposit', iconPNGPath: <IconDeposit />, href: '/' },
    { label: 'Recieve', iconPNGPath: <IconReceive />, href: '/receive' },
    { label: 'Send', iconPNGPath: <IconSendTile />, href: '/send' },
  ]
  const coins = [
    { label: 'USDC', icon: <IconUSDC size={'$2.5'} />, balance: 3.4567 },
    { label: 'Ethereum', icon: <IconEthereum size={'$2.5'} />, balance: 1.25696 },
    { label: 'Send', icon: <IconSend size={'$2.5'} />, balance: 12500123 },
  ]
  const balanceViewButtons = [
    { label: 'Ethereum', onPress: () => {} },
    { label: 'Cards', onPress: () => {} },
  ]
  const transactions = [
    {
      id: 1,
      user: {
        sendTag: 'ethantree',
      },
      type: 'inbound',
      amount: 200,
      currency: 'USDT',
      amountInUSD: 199.98,
      created_on: '',
    },
    {
      id: 2,
      user: {
        sendTag: 'You',
      },
      type: 'outbound',
      amount: 1,
      currency: 'ETH',
      amountInUSD: 1985.56,
      created_on: '',
    },
    {
      id: 3,
      user: {
        sendTag: 'You',
      },
      type: 'outbound',
      amount: 1,
      currency: 'ETH',
      amountInUSD: 1985.56,
      created_on: '',
    },
    {
      id: 4,
      user: {
        sendTag: 'You',
      },
      type: 'outbound',
      amount: 1,
      currency: 'ETH',
      amountInUSD: 1985.56,
      created_on: '',
    },
  ]
  const balanceDetails = [
    {
      currency: 'Ethereum',
      symbol: 'eth',
      balance: 1.45,
    },
    {
      currency: 'USDC',
      symbol: 'usdc',
      balance: 125,
    },
    {
      currency: 'SEND',
      symbol: 'send',
      balance: 71454457,
    },
    {
      currency: 'SEND',
      symbol: 'send',
      balance: 4412,
    },
    {
      currency: 'USDC',
      symbol: 'usdc',
      balance: 2.0,
    },
  ]
  const navigateToScreen = (href: string) => {
    window.location.href = href
  }
  return (
    <>
      <MainLayout scrollable={true}>
        <YStack
          $gtLg={{ width: 600 }}
          $sm={{ width: '100vw' }}
          $gtSm={{ width: '100vw' }}
          ai={'center'}
        >
          {/* Balance Card */}
          <XStack w={'100%'} jc={'center'} borderColor={separatorColor} borderBottomWidth={1}>
            <XStack w={'90%'} zIndex={4}>
              <YStack mx={'$3'} py={'$11'}>
                <YStack jc={'center'} gap={'$6'}>
                  <Paragraph
                    fontSize={'$4'}
                    zIndex={1}
                    color={'$color12'}
                    textTransform={'uppercase'}
                  >
                    [ Total Balance ]
                  </Paragraph>
                  <XStack style={{ color: 'white' }} gap={'$2.5'}>
                    <Paragraph
                      color={'$color12'}
                      fontSize={96}
                      fontWeight={'500'}
                      lineHeight={'$12'}
                      zIndex={1}
                    >
                      {USDollar.format(9489).replace('$', '').split('.')[0]}
                    </Paragraph>
                    <Paragraph color={'$color12'} fontSize={'$6'} fontWeight={'500'} zIndex={1}>
                      {'USD'}
                    </Paragraph>
                  </XStack>
                </YStack>
              </YStack>
            </XStack>
          </XStack>
          <XStack w={'90%'} ai={'center'} jc={'space-evenly'} gap={'$4'} pt={'$7'}>
            <Card
              px={'$3.5'}
              py={'$5'}
              width={'100%'}
              backgroundColor={'$primary'}
              borderRadius={'$4'}
            >
              <XStack jc={'space-between'} ai={'center'}>
                <Paragraph fontWeight={'500'} textTransform={'uppercase'} color={'$black'}>
                  Add Funds
                </Paragraph>
                <XStack alignItems={'center'} justifyContent={'center'} zIndex={2}>
                  <IconDeposit size={'$2.5'} color={'$black'} />
                </XStack>
              </XStack>
            </Card>
          </XStack>
          <YStack width={'90%'} gap={'$3.5'} pt={'$6'} pb={'$12'}>
            {coins.map((coin, index) => (
              <XStack
                jc={'space-between'}
                ai={'center'}
                p={'$3.5'}
                borderColor={separatorColor}
                borderBottomWidth={index !== coins.length - 1 ? 1 : 0}
                key={coin.label}
              >
                <XStack gap={'$2'}>
                  {coin.icon}
                  <Paragraph
                    fontSize={'$5'}
                    fontWeight={'500'}
                    textTransform={'uppercase'}
                    color={'$color12'}
                  >
                    {coin.label}
                  </Paragraph>
                </XStack>
                <Paragraph fontSize={'$9'} fontWeight={'500'} color={'$color12'}>
                  {formatAmount(coin.balance, undefined, 3)}
                </Paragraph>
              </XStack>
            ))}
          </YStack>
        </YStack>
      </MainLayout>
    </>
  )
}
