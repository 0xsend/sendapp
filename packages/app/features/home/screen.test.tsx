import { TamaguiProvider, config } from '@my/ui'
import { render, act, screen } from '@testing-library/react-native'
import { HomeScreen } from './screen'

jest.mock('@my/wagmi')

jest.mock('app/routers/params', () => ({
  useNav: jest.fn().mockReturnValue([undefined, jest.fn()]),
}))

jest.mock('solito', () => {
  // console.log('mock solito')
  const mockCreateParam = jest.fn(() => {
    // console.log('createParam in')
    return {
      useParam: jest.fn(() => {
        // console.log('useParam', name, opts)
        return ['test', jest.fn()]
      }),
      useParams: jest.fn(() => {
        // console.log('useParams', name, opts)
        return ['test', jest.fn()]
      }),
    }
  })
  return {
    __esModule: true,
    createParam: mockCreateParam,
  }
})

jest.mock('app/utils/useUserReferralsCount', () => ({
  useUserReferralsCount: jest.fn().mockReturnValue(123),
}))

jest.mock('app/utils/useSendAccountBalances', () => ({
  useSendAccountBalances: jest.fn().mockReturnValue({
    balances: {
      USDC: 500000n,
      SEND: 500000n,
    },
    totalBalance: () => 1000000n,
  }),
}))
jest.mock('@tamagui/tooltip', () => ({
  ...jest.requireActual('@tamagui/tooltip'),
  TooltipGroup: ({ children }: { children: React.ReactNode }) => <>{children}</>,
}))

jest.mock('solito', () => ({
  createParam: jest
    .fn()
    .mockReturnValue({ useParam: jest.fn().mockReturnValue([undefined, jest.fn()]) }),
}))

jest.mock('app/routers/params', () => ({
  useRootScreenParams: jest.fn().mockReturnValue([{ nav: 'home', token: undefined }, jest.fn()]),
}))

jest.mock('app/features/home/utils/useTokenActivityFeed')

jest.mock('app/utils/supabase/useSupabase', () => ({
  useSupabase: jest.fn().mockReturnValue({
    rpc: jest.fn().mockReturnValue({
      abortSignal: jest.fn().mockReturnValue({
        data: [
          {
            send_id_matches: [],
            tag_matches: [
              {
                send_id: 3665,
                tag_name: 'test',
                avatar_url: 'https://avatars.githubusercontent.com/u/123',
              },
            ],
            phone_matches: [],
          },
        ],
        error: null,
      }),
    }),
  }),
}))

jest.mock('app/utils/send-accounts', () => ({
  useSendAccount: jest.fn().mockReturnValue({
    data: {
      avatar_url: 'https://avatars.githubusercontent.com/u/123',
      name: 'test',
      about: 'test',
      refcode: 'test',
      tag: 'test',
      address: '0x123',
      phone: 'test',
      chain_id: 1,
      is_public: true,
      sendid: 1,
      all_tags: ['test'],
    },
  }),
}))

import { usePathname } from 'expo-router'
// @ts-expect-error mock
usePathname.mockReturnValue('/')
test('HomeScreen', async () => {
  jest.useFakeTimers()
  render(
    <TamaguiProvider defaultTheme={'dark'} config={config}>
      <HomeScreen />
    </TamaguiProvider>
  )
  await act(async () => {
    jest.advanceTimersByTime(2000)
    jest.runAllTimers()
  })
  expect(screen.toJSON()).toMatchSnapshot()
})
