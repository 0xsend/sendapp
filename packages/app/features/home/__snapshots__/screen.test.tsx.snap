// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "display": "none",
      }
    }
  >
    <View
      accessibilityRole="button"
      cancelable={true}
      disabled={false}
      focusVisibleStyle={{}}
      focusable={[Function]}
      minPressDuration={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 100000,
          "borderBottomRightRadius": 100000,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 100000,
          "borderTopRightRadius": 100000,
          "borderTopWidth": 1,
          "display": "none",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "height": 44,
          "justifyContent": "center",
          "maxHeight": 44,
          "maxWidth": 44,
          "minHeight": 44,
          "minWidth": 44,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "right": 0,
          "shadowColor": "rgb(0,0,0)",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "top": -46,
          "width": 44,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="32"
        bbWidth="32"
        fill="none"
        focusable={false}
        height={32}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="hsla(112, 22%, 59%, 1)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "hsla(112, 22%, 59%, 1)",
            },
            {
              "flex": 0,
              "height": 32,
              "width": 32,
            },
          ]
        }
        tintColor="hsla(112, 22%, 59%, 1)"
        vbHeight={24}
        vbWidth={24}
        width={32}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4287016319,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M18 6 6 18"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4287016319,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="m6 6 12 12"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
              ]
            }
            stroke={
              {
                "payload": 4287016319,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "none",
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "gap": 32,
            "justifyContent": "space-between",
            "paddingBottom": 53,
            "paddingLeft": 7,
            "paddingRight": 7,
            "paddingTop": 53,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "width": "100%",
                "zIndex": 4,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "column",
                    "gap": 18,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 10,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#40FB50",
                        "height": 11,
                        "width": 11,
                      }
                    }
                  />
                  <Text
                    cancelable={true}
                    disabled={false}
                    focusable={true}
                    id=":r0:"
                    minPressDuration={0}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onPress={[Function]}
                    onPressIn={[Function]}
                    onPressOut={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "color": "hsl(0, 0%, 90.9%)",
                        "display": "flex",
                        "fontFamily": "System",
                        "fontSize": 16.8,
                        "fontWeight": "500",
                        "lineHeight": 0,
                        "textTransform": "uppercase",
                        "zIndex": 1,
                      }
                    }
                    suppressHighlighting={true}
                    userSelect="none"
                  >
                    Total Balance
                  </Text>
                </View>
                <View
                  style={
                    {
                      "color": "white",
                      "flexDirection": "row",
                      "gap": 10,
                      "marginTop": 13,
                    }
                  }
                >
                  <Text
                    selectable={true}
                    style={
                      {
                        "color": "hsla(112, 22%, 100%, 1)",
                        "fontFamily": "System",
                        "fontSize": 96,
                        "fontWeight": "500",
                        "lineHeight": 119,
                        "zIndex": 1,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    0
                  </Text>
                  <Text
                    selectable={true}
                    style={
                      {
                        "color": "hsla(112, 22%, 100%, 1)",
                        "fontFamily": "System",
                        "fontSize": 21.599999999999998,
                        "fontWeight": "500",
                        "lineHeight": 19,
                        "zIndex": 1,
                      }
                    }
                    suppressHighlighting={true}
                  >
                    USD
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingTop": 18,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityRole="button"
            cancelable={true}
            disabled={false}
            focusVisibleStyle={{}}
            focusable={[Function]}
            minPressDuration={0}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "hsla(125, 96%, 47%, 1)",
                "borderBottomColor": "transparent",
                "borderBottomLeftRadius": 9,
                "borderBottomRightRadius": 9,
                "borderBottomWidth": 1,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 1,
                "borderRightColor": "transparent",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "transparent",
                "borderTopLeftRadius": 9,
                "borderTopRightRadius": 9,
                "borderTopWidth": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                selectable={true}
                style={
                  {
                    "color": "#122023",
                    "fontFamily": "System",
                    "fontSize": 16.8,
                    "fontWeight": "500",
                    "lineHeight": 19,
                    "textTransform": "uppercase",
                  }
                }
                suppressHighlighting={true}
              >
                Deposit
              </Text>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "zIndex": 2,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="32"
                  bbWidth="32"
                  color="#122023"
                  fill="none"
                  focusable={false}
                  height={32}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  strokeWidth="2"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "color": "#122023",
                      },
                      {
                        "flex": 0,
                        "height": 32,
                        "width": 32,
                      },
                    ]
                  }
                  tintColor="#122023"
                  vbHeight={24}
                  vbWidth={24}
                  width={32}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                        "strokeWidth",
                      ]
                    }
                    strokeWidth="2"
                  >
                     
                    <RNSVGRect
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      height="22"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                        ]
                      }
                      rx="11"
                      stroke={
                        {
                          "payload": 4279377955,
                          "type": 0,
                        }
                      }
                      strokeWidth="2"
                      width="22"
                      x="1"
                      y="1"
                    />
                    <RNSVGMask
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      height="16"
                      maskContentUnits={1}
                      maskUnits={1}
                      name="mask0_3996_6472"
                      width="16"
                      x="4"
                      y="4"
                    >
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4292467161,
                            "type": 0,
                          }
                        }
                        height="14.625"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        width="14.625"
                        x="4.5"
                        y="4.5"
                      />
                    </RNSVGMask>
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      mask="mask0_3996_6472"
                    >
                      <RNSVGPath
                        d="M11.3571 12.6429H6V11.3571H11.3571V6H12.6429V11.3571H18V12.6429H12.6429V18H11.3571V12.6429Z"
                        fill={
                          {
                            "type": 2,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#E6E6E6",
            "borderBottomWidth": 1,
            "borderLeftColor": "#E6E6E6",
            "borderLeftWidth": 0,
            "borderRightColor": "#E6E6E6",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "#E6E6E6",
            "borderTopWidth": 0,
            "flex": 1,
            "flexShrink": 0,
            "height": 0,
            "maxHeight": 0,
            "transform": [
              {
                "translateY": -0.5,
              },
            ],
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "none",
              "flexDirection": "column",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "column",
          "gap": 24,
          "paddingBottom": 32,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 32,
          "width": "100%",
        }
      }
    >
      <Text
        accessibilityRole="header"
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 30,
            "fontWeight": "700",
            "lineHeight": 27,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "textAlign": "center",
            "textTransform": "none",
          }
        }
        suppressHighlighting={true}
      >
        Register Your Sendtag Today!
      </Text>
      <View
        style={
          {
            "flexDirection": "column",
            "gap": 13,
            "width": "100%",
          }
        }
      >
        <Text
          selectable={true}
          style={
            {
              "color": "hsla(112, 22%, 100%, 1)",
              "fontFamily": "System",
              "fontSize": 16.8,
              "fontWeight": "400",
              "lineHeight": 19,
            }
          }
          suppressHighlighting={true}
        >
          By registering a Sendtag, you can:
        </Text>
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <Text
            selectable={true}
            style={
              {
                "color": "hsla(112, 22%, 100%, 1)",
                "fontFamily": "System",
                "fontSize": 16.8,
                "fontWeight": "400",
                "lineHeight": 19,
              }
            }
            suppressHighlighting={true}
          >
            1. Send and receive funds using your personalized identifier
          </Text>
          <Text
            selectable={true}
            style={
              {
                "color": "hsla(112, 22%, 100%, 1)",
                "fontFamily": "System",
                "fontSize": 16.8,
                "fontWeight": "400",
                "lineHeight": 19,
              }
            }
            suppressHighlighting={true}
          >
            2. Earn Send It Rewards based on your token balance and referrals
          </Text>
        </View>
        <Text
          selectable={true}
          style={
            {
              "color": "hsla(112, 22%, 100%, 1)",
              "fontFamily": "System",
              "fontSize": 16.8,
              "fontWeight": "400",
              "lineHeight": 19,
            }
          }
          suppressHighlighting={true}
        >
          Join us in shaping the Future of Finance.
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": "auto",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "auto",
          }
        }
      />
    </View>
  </View>
</View>
`;
